tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -1369,28 +1366,57 @@ def test_search_runs_returns_expected_results_with_large_experiment(self):
                 }
                 metrics_list.append(metric)
                 tag = {
-                    'key': "tkey-%s" % i,
-                    'value': "tval-%s" % i,
+                    'key': "tkey_%s" % i,
+                    'value': "tval_%s" % i,
                     'run_uuid': run_id,
                 }
                 tags_list.append(tag)
                 param = {
-                    'key': "pkey-%s" % i,
-                    'value': "pval-%s" % i,
+                    'key': "pkey_%s" % i,
+                    'value': "pval_%s" % i,
                     'run_uuid': run_id,
                 }
                 params_list.append(param)
+            last_metrics_list.append(
+                {
+                    'key': 'mkey_0',
+                    'value': current_run,
+                    'timestamp': 100 * 2,
+                    'step': 100 * 3,
+                    'is_nan': 0,
+                    'run_uuid': run_id,
+                }
+            )
+            current_run += 1
         metrics = pd.DataFrame(metrics_list)
         metrics.to_sql('metrics', self.store.engine, if_exists='append', index=False)
         params = pd.DataFrame(params_list)
         params.to_sql('params', self.store.engine, if_exists='append', index=False)
         tags = pd.DataFrame(tags_list)
         tags.to_sql('tags', self.store.engine, if_exists='append', index=False)
+        pd.DataFrame(last_metrics_list).to_sql(
+            'latest_metrics', self.store.engine, if_exists='append', index=False)
+        return experiment_id, run_ids
+
+    def test_search_runs_returns_expected_results_with_large_experiment(self):
+        """
+        This case tests the SQLAlchemyStore implementation of the SearchRuns API to ensure
+        that search queries over an experiment containing many runs, each with a large number
+        of metrics, parameters, and tags, are performant and return the expected results.
+        """
+        experiment_id, run_ids = self._generate_large_data()
 
         run_results = self.store.search_runs([experiment_id], None, ViewType.ALL, max_results=100)
-        assert len(run_results) > 0
+        assert len(run_results) == 100
         assert set([run.info.run_id for run in run_results]).issubset(set(run_ids))
 
+    def test_search_runs_correctly_handles_filtering(self):
```

Comment:
Can you rename this to something like `test_search_runs_correctly_filters_large_data`? There are other tests that verify filter performance on smaller data, so it's best to have a more distinct test name.